### DISCUSSION TITLE
Get disconnected in an interval of 60s to 90s

### DISCUSSION THREAD

**Discussion by tootai** [Ylianst/MeshCentral#6114] [Issues] [answered]
Hi,
I connect to the desktop of an agent -here a MBP Sonoma- which is outside the server lan, and after the laps of time of 60 to 90s or so I get disconnected, agent still showed as connected but tab Desktop and Terminal disappeared.
To get them back, I go to Console tab => agent action => upload recovery kernel + upload  default server kernel. After this, both tabs are coming back, I can connect again till the next disconnection after the same amount of time :(

Settings are
"settings": {
    "cert": "support.TLD",
    "_WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword",
    "trustedProxy": "fd90:1:2:3::1",
    "tlsOffload": "fd90:1:2:3::1",
    "port": 4430,
    "aliasPort": 443,
    "_redirPort": 8000,
    "_redirAliasPort": 80,
    "_agentPing": 55,
    "_agentPong": 25
  }

MC 1.1.22

Thanks for any hint, Daniel

**Comment by si458** [comment, marked as answer]
are you using IPv6 only? (guessing from the trustedProxy value)
then set `agentPong: 25` (so remove the `_`) then restart meshcentral

**Reply by tootai** [reply]
OK, that did the trick, thanks

**Reply by si458** [reply]
basically the agentPong, send a `ping` packet every x seconds to keep the websocket open,
you can also use `agentPing` which sends a `ping` packet every x seconds to keep the websocket open AND also expects the remote device to reply with `pong` when it gets the `ping` packets, and if not `disconnect` the agent
but generally `agentPong` is the best one to use

edit: in theory thw websockets should have there own `ping/pong` built in method but sometimes it doesnt work,
hense why we added `agentPong` and `agentPing` so it does a custom `ping/pong` as well as the built in method

**Reply by tootai** [reply]
Thanks for your clear explanations, as usually you're on the top, keep this good work.

**Reply by si458** [reply]
Donations / Amazon Wish List always accepted! https://www.si458.co.uk/2024/01/05/donation/
